docs/debug-with-host.md

```diff
@@ -0,0 +1,55 @@
+# Guide on debugging Core Tools with the Azure Functions Host
+
```

Comment:
The changes here don't seem to be directly related to this PR.

Reply:
Moved to a new PR: #4546